tools/dkcm/clerk/clerk.go

```diff
@@ -0,0 +1,121 @@
+/*
+Copyright 2020 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package clerk
+
+import (
+	"database/sql"
+	"fmt"
+	"time"
+
+	"k8s.io/apimachinery/pkg/util/uuid"
+	"knative.dev/test-infra/pkg/mysql"
+)
+
+// Cluster stores a row in the "Cluster" db table
+type Cluster struct {
+	ClusterID string
+	ProjectID string
+	Status    string
+	Zone      string
+	Nodes     int
+	NodeType  string
+}
+
+// Request stores a request for the "Request" table
+type Request struct {
+	accessToken string
+	requestTime time.Time
+	Zone        string
+	Nodes       int
+	NodeType    string
+	ProwJobID   string
+}
+
+// ClusterConfig is the struct to return to Prow once the cluster is available
+type ClusterConfig struct {
+	ClusterName string
+	ProjectID   string
+	Zone        string
+}
+
+type ClerkOperations interface {
```

Comment:
Should you also have functions like:
- NewRequest()
- MapRequest()
- ListRequest()

Also I think before cluster started creation, it should be marked started in clerk, so that it will be counted